src/xml-properties-mixin.js

```diff
@@ -0,0 +1,180 @@
+//
+// key: value objects
+// properties.foo = { bar: 5, baz: 'bat' } ==> <foo bar="5", baz="bat"/>
+//
+// array elements
+// properties.push('foo', {bar: 5})
+// properties.push('foo', {bar: 6})
+// ==> <foo bar="5"/><foo bar="6"/>
+//
+
+let XmlPropertiesMixin = (superclass) => class extends superclass {
+  //
+  // make a new instance, given a root element and a propertiesTag
+  //
+  constructor(root, { propertiesTag } = {}) {
+    if (!propertiesTag) {
+      throw new Error(`need propertiesTag for XmlProperties`);
+    }
+    super(...arguments);
+    // accept either an XmlProperties object (with a .$root property) or a dom element
+    this.$root = root.$root || $(root);
+    if (!this.$root.length) {
+      throw new Error(`need root for XmlProperties`);
+    }
+    this.propertiesTag = propertiesTag;
+  }
+  //
+  // get the $Pr element where our properties are stored
+  //
+  get $Pr() {
+    return this.$root.find(`> ${this.toSelector(this.propertiesTag)}`);
+  }
+  //
+  // return the html of the $Pr element
+  //
+  html() {
+    return $.html(this.$Pr);
+  }
+  //
+  // return our $Pr element, creating it as a child of our root element if necessary
+  //
+  ensurePr() {
+    let $prop = this.$Pr;
+    if (!$prop.length) {
+      $prop = $(`<${this.ensureNamespace(this.propertiesTag)}/>`);
+      this.$root.prepend($prop);
+    }
+    return $prop;
+  }
+  //
+  // find the $element corresponding to a given tag in our $Pr element
+  //
+  findProperty(tag) {
+    let $prop = this.$Pr.find(`> ${this.toSelector(tag)}`);
+    if ($prop.length) {
+      return $prop;
+    }
+  }
+  //
+  // return an $element for the given tag, creating it as a child of our $Pr
+  // element if necessary
+  //
+  ensureProperty(tag) {
+    let $prop = this.findProperty(tag);
+    if (!$prop) {
+      $prop = $(`<${this.ensureNamespace(tag)}/>`);
+      this.ensurePr().append($prop);
+    }
+    return $prop;
+  }
+  //
+  // remove the $element for the given tag
+  //
+  removeProperty(tag) {
+    let $prop = this.findProperty(tag);
+    if ($prop) {
+      $prop.remove();
+    }
+  }
+  //
+  // store the keys/values in the given attrs hash as attributes on a dom element
+  //
+  setAttrs(el, attrs = {}) {
+    Object.keys(attrs).forEach((key) => {
+      $(el).attr(key, attrs[key]);
+    });
+  }
+  //
+  // get an array of pojo values for a given tag
+  //
+  getArray(tag) {
+    let $vals = this.findProperty(tag);
+    if (!$vals) {
+      return [];
+    }
+    return $vals.toArray().map((item) => $(item).attr());
+  }
+  //
+  // set an array of pojo values for a given tag
+  //
+  setArray(tag, attrList = [], $Pr = this.ensurePr()) {
+    // remove existing elements
+    this.removeProperty(tag);
+    // set the new elements
+    attrList.forEach((attrs) => this.push(tag, attrs, $Pr));
+  }
+  //
+  // add one array-item having the given attributes for the given tag
+  //
+  push(tag, attrs = {}, $Pr = this.ensurePr()) {
```

Comment:
The main thing I'd like to see is that each of the type-specific methods clearly indicate the type they are operating on. I can't think of something else somebody would think `push` does, but it might not be clear what it actually does do, and including the words `push` and `array` I think will make it immediately clear (and also clear that whatever you're calling `push()` on isn't just a regular old array). So `pushToArray` satisfies those considerations.